### ISSUE TITLE
Cannot mock function with generic type

### ISSUE THREAD

**Issue by AlexandreBrown** [mockative/mockative#37]
Hello,

I am trying to mock a function with generic type but I'm not certain if that is possible or how to do it exactly.

Here is the function signature from the interface I'm trying to mock :  
```kotlin
suspend fun <T> getObject(input: GetObjectRequest, block: suspend (GetObjectResponse) -> T): T
```
In my implementation, `getObject<File>` is being called.  

My failed attempt to mock it :  
```kotlin
given(s3Client).suspendFunction(s3Client::getObject, fun2<GetObjectRequest, suspend (GetObjectResponse) -> File>())
		.whenInvokedWith(any(), any())
		.then {
			File("Dummy")
		}
```
Error :  
![image](https://user-images.githubusercontent.com/26939775/172060406-ea794fde-33ab-465c-8e8d-fc05533f5389.png)  
`None of the following functions can be called with the arguments supplied.`

**Comment by Nillerr** [comment]
Hi @AlexandreBrown

I'll have a look at what could be done to improve this specific use case. For now, you may be able to use the string-based mocking function using something like this:

```kotlin
given(s3Client).suspendFunction("getObject")
    .whenInvokedWith(any(), any())
    .then { ... }
```

**Comment by AlexandreBrown** [comment]
@Nillerr For me the untyped callable references `suspendFunction(String)` is not defined (using Mockative 1.2.3).  
Any idea?

**Comment by Nillerr** [comment]
You're right 😅 I referenced a function that's no longer there. I must have removed it at some point, or perhaps just never gotten around to implementing it.

Tomorrow I'll have a look as to whether or not I can reintroduce the stringly-typed variant, as well as whether I can better support the type of function you're trying to mock.

**Comment by Nillerr** [comment]
Hi again @AlexandreBrown 

I have implemented a test for your example and it turns out the issue is you're statically referencing the `getObject` function of `S3Client` like so: `S3Client::getObject`, where Mockative is designed such that you should be referencing the instance-scoped function `s3Client::getObject` (notice the lowercase `s`, referencing the _property_).

Let me know if this addresses your issue 🙂

**Comment by AlexandreBrown** [comment]
@Nillerr Sorry about that, it was a typo in the issue description, I use Mockative by using the instance-scoped function `s3Client::getObject`.  
Updated the issue description.

**Comment by Nillerr** [comment]
I published [v1.2.5](https://github.com/mockative/mockative/releases/v1.2.5) earlier today. Let me know if this resolves your issue 🙂

**Comment by AlexandreBrown** [comment]
@Nillerr Thanks for the update, this fixed my issue and it works now! :)
